lib-persistence-manager/persistencemgr/redis.go

```diff
@@ -1194,13 +1195,13 @@ func (p *ConnPool) DeleteDeviceSubscription(index, hostIP string) error {
 // 1. index is the name of the index to be created
 // 2. key and value are the key value pair for the index
 func (p *ConnPool) UpdateDeviceSubscription(index, hostIP, location string, originResources []string) error {
-	_, err := p.GetDeviceSubscription(index, hostIP+"*")
+	_, err := p.GetDeviceSubscription(index, hostIP+"[^0-9]*")
```

Comment:
Shouldn't we use '$' to denote the end of string to match exact string?

Reply:
tried $ to match the exact string, it's not working in the sorted set redis